src/poetry/core/masonry/metadata.py

```diff
@@ -107,11 +107,12 @@ def from_package(cls, package: ProjectPackage) -> Metadata:
 
         if package.urls:
             for name, url in package.urls.items():
-                if name.lower() == "homepage" and meta.home_page == url:
+                name_lower = name.lower()
+                if name_lower == "homepage" and meta.home_page == url:
                     continue
-                if name == "repository" and url == package.urls["Repository"]:
+                if name_lower == "repository" and url == package.urls.get(name):
```

Comment:
is this not always definitely true?  because `name` and `url` are a pair from `package.urls.items()`.

I am not sure what the intent of this code was but this variation seems pointless?

Reply:
perhaps the original code was just a roundabout way of reading `package.repository_url` - and doing that directly is a better fix?

Reply:
yeah i think the original code wasn't well written, i'll simplify it 👍🏼